avro-serializer/src/main/java/io/confluent/kafka/serializers/KafkaAvroDeserializer.java

```diff
@@ -53,6 +54,11 @@ public Object deserialize(String s, byte[] bytes) {
     return deserialize(bytes);
   }
 
+  /**
+   * Pass a reader schema to get an Avro projection
+   */
+  public Object deserialize(String s, byte[] bytes, Schema readerSchema) { return deserialize(bytes, readerSchema); }
```

Comment:
`most people don't drive the deserializer manually`

You're right.  We may want to consider projections to be an advanced feature.

If you think it's useful, we could allow three different behaviors:
* Existing behavior
* Statically configured projection
  * We could add a config param called "avro.reader.schema" which you statically set to the one you need for cases where there's a single projection per Decoder/Deserializer/Converter.
* Manual override
  * For more complex cases where you need multiple projections, you can pass a reader schema per call.